astroquery/heasarc/core.py

```diff
@@ -135,290 +455,293 @@ def query_object_async(self, object_name, mission, *,
+            supported.
+        profile : str
+            Profile to use to identify yourself to the cloud provider
+            (usually in ~/.aws/config).
         """
-        # Convert the coordinates to FK5
-        c = commons.parse_coordinates(position).transform_to(coordinates.FK5)
-        kwargs['coordsys'] = 'fk5'
-        kwargs['equinox'] = 2000
-
-        # Generate the request
-        # Fixed string representation of coordinates ensures that request payload
-        # does not depend on python/astropy version for the same input coordinates
-        request_payload = self._args_to_payload(
-            mission=mission,
-            entry=f"{c.ra.degree:.10f},{c.dec.degree:.10f}",
-            radius=u.Quantity(radius),
-            **kwargs
-        )
 
-        # Return payload if requested
-        if get_query_payload:
-            return request_payload
+        try:
+            import boto3
+            import botocore
+        except ImportError:
+            raise ImportError(
+                'The cloud feature requires the boto3 package. '
+                'Install it first.'
+            )
+
+        if profile is None:
+            log.info('Enabling annonymous cloud data access ...')
+            config = botocore.client.Config(
+                signature_version=botocore.UNSIGNED)
+            self.s3_resource = boto3.resource('s3', config=config)
+
+        elif isinstance(profile, bool) and not profile:
+            # profile is False, use system env credentials
+            self.s3_resource = boto3.resource('s3')
 
-        # Submit the request
-        return self.query_async(request_payload, cache=cache)
+        else:
+            log.info('Enabling cloud data access with profile: {profile} ...')
+            session = boto3.session.Session(profile_name=profile)
+            self.s3_resource = session.resource(service_name='s3')
 
-    def _old_w3query_fallback(self, content):
-        # old w3query (such as that used in ISDC) return very strange fits, with all ints
+        self.s3_client = self.s3_resource.meta.client
 
-        fits_content = fits.open(BytesIO(content))
+    def download_data(self, links, host='heasarc', location='.'):
+        """Download data products in links with a choice of getting the
+        data from either the heasarc server, sciserver, or the cloud in AWS.
 
-        for col in fits_content[1].columns:
-            if col.disp is not None:
-                col.format = col.disp
-            else:
-                col.format = str(col.format).replace("I", "A")
 
-        tmp_out = BytesIO()
-        fits_content.writeto(tmp_out)
-        tmp_out.seek(0)
+        Parameters
+        ----------
+        links : `astropy.table.Table`
+            The result from get_links
+        host : str
+            The data host. The options are: heasarc (defaul), sciserver, aws.
+            If host == 'sciserver', data is copied from the local mounted
```

Comment:
this is very specific corner case only for sciserver. Will it work for smce, too?

Reply:
I agree this is specific to Sciserver, we have many users there and this make it data access efficient for those users. The option will work for fornax